### PR TITLE
exit if required library/file is not found

### PR DESCRIPTION
**Pull request by minshao**

### CODE REVIEW

File: src/pcre_check/CMakeLists.txt [hunk 1]
```diff
@@ -6,20 +6,47 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-exit-time-destructors")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-parameter")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-padded")
 
-find_package(Lua REQUIRED) # TODO revisit!!!
+if (REMATCHCOMP_LIB AND HYPERSCAN_LIB AND PCRE2_LIB)
+  set(CONTINUE TRUE)
+endif()
+
+if (CONTINUE)
+  find_package(Lua) # TODO revisit!!!
```
Comment (msk) [inline src/pcre_check/CMakeLists.txt]: I don't think pcre_check uses Lua. This line was probably just copied without review.

File: src/pcre_check/CMakeLists.txt [hunk 2]
```diff
@@ -6,20 +6,47 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-exit-time-destructors")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-parameter")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-padded")
 
-find_package(Lua REQUIRED) # TODO revisit!!!
+if (REMATCHCOMP_LIB AND HYPERSCAN_LIB AND PCRE2_LIB)
+  set(CONTINUE TRUE)
```
Comment (msk) [inline src/pcre_check/CMakeLists.txt]: How about testing everything, including PkgConfig and auto-opt, and set `CONTINUE` after then?

**Review by msk — CHANGES_REQUESTED** [review]

File: src/pcre_check/CMakeLists.txt (hunk 2 above)
Comment (minshao) [reply src/pcre_check/CMakeLists.txt]: sure, in that case, it will be not be able to early exit though, is that OK?

Comment (msk) [reply src/pcre_check/CMakeLists.txt]: I think so. It will consume more time during the configure stage, but it shouldn't take long.

**Review by msk — APPROVED** [review]

**Pull request merged**